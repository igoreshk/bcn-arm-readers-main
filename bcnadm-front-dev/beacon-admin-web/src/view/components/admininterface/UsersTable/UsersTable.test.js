import React from 'react';
import { shallow } from 'enzyme';

import { UsersTable } from './index';

const props = {
  translate: jest.fn(),
  showLoadingScreen: jest.fn(),
  hideLoadingScreen: jest.fn(),
  history: {},
  updated: false
};

const filteredUsers = [
  { entityId: 3, name: 'test3' }
];

const wrapper = shallow(<UsersTable {...props} />);

describe('UserTable', () => {
  const loadData = jest.spyOn(UsersTable.prototype, 'loadData');
  const showLoadingScreen = jest.spyOn(wrapper.instance().props, 'showLoadingScreen');

  it('Checks that props are passed properly', () => expect(wrapper.instance().props).toEqual(props));

  it('Check showLoadingScreen was called from componentWillMount and componentDidUpdate life hooks', () => {
    wrapper.setProps({ updated: true });
    expect(showLoadingScreen).toHaveBeenCalledTimes(2);
  });

  it('Check loadData was called from componentDidUpdate life hook', () => expect(loadData).toHaveBeenCalled());

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());

  it('Checks if history is called with correct arguments', () => {
    wrapper.setState({
      filteredUsers
    });
    wrapper.find('.user').forEach((node) => {
      node.simulate('click');
      expect(wrapper.instance().props.history.push).toHaveBeenCalledWith(filteredUsers[0].entityId);
    });
  });
});
