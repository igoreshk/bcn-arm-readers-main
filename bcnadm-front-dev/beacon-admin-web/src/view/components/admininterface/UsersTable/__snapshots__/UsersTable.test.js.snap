// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable Check matching snapshot 1`] = `""`;
